yamllint/config.py

```diff
@@ -96,6 +102,12 @@ def parse(self, raw_content):
             self.ignore = pathspec.PathSpec.from_lines(
                 'gitwildmatch', conf['ignore'].splitlines())
 
+        if 'yaml-files' in conf:
+            if not isinstance(conf['yaml-files'], list):
```

Comment:
This would crash yamllint (instead of displaying a nice error) in case of malformed conf like:
```yaml
yaml-files:
  - file: name/*.yaml
```
Can you add `and all(isinstance(i, str) for i in conf['yaml-files'])` or something equivalent?

Reply:
added